### PR TITLE
VB-4528 - New endpoint to get a single session

### PR DESCRIPTION
**Pull request by adaviesMOJ**
## What does this pull request do?

Allows the frontend to retrieve a single visit session and all it's details.

## JIRA Link
https://dsdmoj.atlassian.net/browse/VB-4528

### CODE REVIEW

**Review by hutcheonb-moj — DISMISSED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/session/GetSessionTest.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    // Then
+    val returnResult = responseSpec.expectStatus().isOk
+      .expectBody()
+    val visitSession = getResults(returnResult)
+
+    Assertions.assertThat(visitSession.sessionTemplateReference).isEqualTo(sessionTemplate.reference)
+    Assertions.assertThat(visitSession.openVisitCapacity).isEqualTo(sessionTemplate.openCapacity)
+    Assertions.assertThat(visitSession.openVisitBookedCount).isEqualTo(1)
+  }
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/session/GetSessionTest.kt:56]: do we need a -ve test where session date passed is incorrect?

**Review by dhirajshettymoj — COMMENTED** [review]
minor comment

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/session/GetSessionTest.kt (hunk 1 above)
Comment (adaviesMOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/session/GetSessionTest.kt:56]: Added test to handle not finding resource, thank you

**Review by dhirajshettymoj — DISMISSED** [review]
LGTM

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**